packages/veritone-react-common/src/components/MediaPlayer/VideoSource.js

```diff
@@ -29,6 +33,51 @@ export default class VideoSource extends React.Component {
 
   UNSAFE_componentWillReceiveProps(nextProps) {
     const { video, src, streams, disablePreload } = nextProps;
+    this.loadVideo(video, src, streams, disablePreload);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const {
+      video,
+      src,
+      streams,
+      disablePreload,
+      currentTime,
+      actions,
+    } = this.props;
+    if (
+      !disablePreload &&
+      video &&
+      this.state.src &&
+      this.state.src !== prevState.src
+    ) {
+      video.load();
+    }
+    if (this.state.expiredSignedUri !== prevState.expiredSignedUri) {
+      this.setState({
+        streamUri: undefined,
```

Comment:
Just wondering how to get the new signed url when streamUri is expired? It seems streamUri is from streams. will streams be updated from somewhere?

Reply:
I think the stream URI remains the same, it will have a link like this
`"uri": "https://api.stage.us-gov-2.veritone.com/v3/stream/2360000033/dash.mpd",`
Once the media player loads the stream, it will sign new one and expired after 3 hours.